spec/ic.did

```diff
@@ -35,6 +35,68 @@ type http_request_error = variant {
   conn_timeout;
 };
 
+type satoshi = nat64;
+
+type bitcoin_network = variant {
+  mainnet;
+  testnet;
+};
+
+type bitcoin_address = text;
+
+type block_hash = blob;
+
+type outpoint = record {
+  txid : blob;
+  vout : nat32
+};
+
+type utxo = record {
+  outpoint: outpoint;
+  value: satoshi;
+  height: nat32;
+};
+
+type pagination = record {
+  tip_block_hash : block_hash;
+  offset: opt nat32;
```

Comment:
Note that, for performance reasons, the `offset` will likely be a blob rather than a `nat32`. I'll update the PR here once I have sufficient clarity.

Reply:
Can you elaborate? Where does the impact come from and how big is it?

Reply:
@ielashi Can you say more?

Reply:
BTW, the official Candid guidance by @rossberg is to not use `nat32` in Candid unless you really want a bit vector (e.g. for crypto). `nat` is usually _smaller_ on the wire, too.

Maybe the performance of `blob` benefit @ielashi alludes to is that we can avoid that discussion :-D

Reply:
I don't think so. Even if such a list existed, we cannot efficiently find and remove elements. It's also more complicated due to the following point that I briefly brushed over:

> There's [...] an additional challenge related to storing a list of unbounded size in stable memory, but it's not too relevant for this discussion so I'll leave out the details.

Even if we had a `SerializedList` implementation, it would make the solution more complex. This map is stored in stable memory where we manage the memory ourselves. With our approach of a `BTreeSet` every entry is bounded in size, and we need is a simple constant-size allocator for managing memory. On the other hand, a `SerializedList` would be unbounded in size, and would require implementing a dynamic memory allocator, and that introduces a lot of additional complexity.